app/lib/composegenerator/v3/generate.py

```diff
@@ -85,11 +85,12 @@ def createComposeConfigFromV3(app: dict, nodeRoot: str):
     newApp = configureIps(newApp, networkingFile, envFile)
     # This is validated earlier
     for container in newApp.containers:
-        container.ports = container.requiredPorts
+        for tcpPort in container.requiredPorts:
+            container.ports.append("{}:{}".format(tcpPort, tcpPort))
         del container.requiredPorts
     for container in newApp.containers:
         for udpPort in container.requiredUdpPorts:
-            container.ports.append("{}/udp".format(udpPort))
+            container.ports.append("{}/udp:{}/udp".format(udpPort, udpPort))
```

Comment:
To document these changes somewhat, I thought if the port on the inside and outside was the same, directly having the port in the ports section without a target port would be enough, but it seems like docker needs this explicit definition.